src/Approximations/iterative_refinement.jl

```diff
@@ -17,12 +17,12 @@ Type that represents a local approximation in 2D.
 
 The criteria for being refinable are determined in the method `new_approx`.
 """
-struct LocalApproximation{N<:Real}
-    p1::Vector{N}
-    d1::Vector{N}
-    p2::Vector{N}
-    d2::Vector{N}
-    q::Vector{N}
+struct LocalApproximation{N<:Real, VN<:AbstractVector{N}}
+    p1::VN
+    d1::VN
+    p2::VN
+    d2::VN
+    q::VN
```

Comment:
Are you sure that all vectors should have the same type? For instance, `q` comes from
`q = element(intersection(Line(d1, dot(d1, p1)), Line(d2, dot(d2, p2))))`.

Reply:
No, i'm not sure if we need to add more types to fix https://github.com/JuliaReach/LazySets.jl/issues/1874, but this is a first step in that direction. This change shouldn't break anything, because in  `master`, all vectors also have the same type. In this branch, all vectors also have the same type. `VN` is just a parameter so we can use static vectors.

Reply:
I agree that it is still an improvement over the current state :+1: